### PR TITLE
feat: [CDS-80416]: fix static schema for steps in custom stage

### PR DESCRIPTION
**Pull request by shalini-agr**
Added yamls for parallel and group type for steps and fixed schema for http and shell script steps as per the RFC

### CODE REVIEW

File: v1/pipeline/steps/custom/step-group-node.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+title: StepGroupNode
+type: object
+required:
+  - spec
+  - type
+properties:
+  desc:
+    type: string
+    desc: This is the description for ShellScriptStepNode
```
Comment (archit-harness) [inline v1/pipeline/steps/custom/step-group-node.yaml:9]: this description is wronng, plus field here should be description isnt?

Comment (shalini-agr) [reply v1/pipeline/steps/custom/step-group-node.yaml]: yes, done

File: v1/pipeline/common/policy-config.yaml [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 properties:
-  policySets:
+  policy_sets:
     oneOf:
     - type: array
       items:
         type: string
     - type: string
-      pattern: ^<\+input>((\.)((executionInput\(\))|(allowedValues|default|regex)\(.+?\)))*$
+      pattern: (<\+.+>.*)
```
[the remark this answers is not in the document]
Comment (shalini-agr) [reply v1/pipeline/common/policy-config.yaml]: done

**Review by archit-harness — DISMISSED** [review]

File: v1/pipeline/steps/custom/http-step-info.yaml [hunk 3]
```diff
@@ -22,49 +22,15 @@ allOf:
...
-    outputVariables:
+    output_vars:
       type: array
       items:
         oneOf:
         - $ref: ../../common/number-ng-variable.yaml
         - $ref: ../../common/secret-ng-variable.yaml
         - $ref: ../../common/string-ng-variable.yaml
-    requestBody:
+    request_body:
```
Comment (hextechpal) [inline v1/pipeline/steps/custom/http-step-info.yaml:32]: just call this body

**Review by hextechpal — DISMISSED** [review]

File: v1/pipeline/steps/custom/http-step-info.yaml (hunk 3 above)
Comment (shalini-agr) [reply v1/pipeline/steps/custom/http-step-info.yaml]: done

**Review by archit-harness — APPROVED** [review]

**Review by hextechpal — APPROVED** [review]

**Pull request merged**